gremlin-server/src/test/java/org/apache/tinkerpop/gremlin/server/GremlinDriverIntegrateTest.java

```diff
@@ -379,7 +380,7 @@ public void shouldEventuallySucceedOnSameServerWithScript() throws Exception {
             fail("Should not have gone through because the server is not running");
         } catch (Exception i) {
             final Throwable root = ExceptionUtils.getRootCause(i);
-            assertThat(root, instanceOf(TimeoutException.class));
+            assertThat(root, instanceOf(ConnectException.class));
```

Comment:
This is a nicer exception that the "timeout" - cool.

Reply:
Yeah, the intention is that the users (application code) should be able to distinguish between retriable and non-retriable errors. They could write retry strategy on the timeout error but the connect error might signify something which will not fix by retrying.